README.md

```diff
@@ -21,49 +73,409 @@ that's been made default from that version forward.
+if you do need to customise this is where you would go.
+
+
+### Loaders
+
+Webpack enables use of loaders to preprocess files. This allows you to
+bundle any static resource way beyond JavaScript. All base loaders
+that ships with webpacker are located inside `config/webpack/loaders`.
+
+If you want to add a new loader, for example to process `json` files via webpack:
+
+```
+yarn add json-loader
+```
+
+and create a `json.js` file inside `loaders` directory:
+
+```js
+module.exports = {
+  test: /\.json$/,
+  use: 'json-loader'
+}
+```
+
+Now if you import('data.json') any `.json` files inside your javascript
+they will be processed using `json-loader`. Voila!
+
+
+### Paths
+
+By default, webpacker ships with simple conventions for where the javascript
+app files and compiled webpack bundles will go in your rails app,
+but all these options are configurable from `config/webpacker.yml` file.
+
+The configuration for what Webpack is supposed to compile by default rests
+on the convention that every file in `app/javascript/packs/*`**(default)**
+or whatever path you set for `source_entry_path` in the `webpacker.yml` configuration
+is turned into their own output files (or entry points, as Webpack calls it).
+
+Suppose you want to change the source directory from `app/javascript`
+to `frontend` this is how you would do it:
+
+```yml
+# config/webpacker.yml
+source_path: frontend
+source_entry_path: packs
+public_output_path: assets/packs
+```
+
+Similary you can also control and configure `webpack-dev-server` settings from
+`config/webpacker.yml` file:
+
+```yml
+# config/webpacker.yml
+development:
+  dev_server:
+    host: 0.0.0.0
+    port: 8080
+    https: false
+```
+
+
+### Babel
+
+Webpacker ships with [babel](https://babeljs.io/) - a JavaScript compiler so,
+you can use next generation JavaScript, today. The Webpacker installer sets up a default standard `.babelrc` file in your app root, which will work great in most cases
+because of [babel-env-preset](https://github.com/babel/babel-preset-env).
+
+Following ES6/7 features are supported out of the box:
+
+* Async/await.
+* Object Rest/Spread Properties.
+* Exponentiation Operator.
+* Dynamic import() - useful for route level code-splitting
+* Class Fields and Static Properties.
+
+We have also included [babel polyfill](https://babeljs.io/docs/usage/polyfill/)
+that includes a custom regenerator runtime and core-js.
+
+
+### Post-Processing CSS
+
+Webpacker out-of-the-box provides CSS post-processing using
+[postcss-loader](https://github.com/postcss/postcss-loader)
+and the installer sets up a standard `.postcssrc.yml` file in your app root
+with standard plugins.
+
+```yml
+plugins:
+  postcss-smart-import: {}
+  precss: {}
+  autoprefixer: {}
+```
+
+
+### CDN
+
+Webpacker out-of-the-box provides CDN support using `ASSET_HOST` environment variable
+set within your Rails app. You don't need to do anything extra, it just works. Voila!
```

Comment:
You don't have to set or know about `ASSET_HOST`. It's just what our `bin/` scripts use internally. Webpacker uses your app's `config.action_controller.asset_host`. http://guides.rubyonrails.org/asset_pipeline.html#cdns or http://guides.rubyonrails.org/configuring.html#rails-general-configuration